app/src/main/java/com/odysee/app/GoLiveActivity.java

```diff
@@ -0,0 +1,600 @@
+            textChannelError.setText(getString(R.string.channel_error_pending));
+            textChannelError.setVisibility(View.VISIBLE);
+            return false;
+        }
+
+        // Disabled due to lack of server side checking for now
+//        double effectiveAmount = Double.parseDouble(channel.getMeta().getEffectiveAmount());
+//        if (effectiveAmount < MIN_STREAM_STAKE) {
+//            textChannelError.setText(getString(R.string.channel_error_need_minimum_credits,
+//                    MIN_STREAM_STAKE, channel.getName()));
+//            textChannelError.setVisibility(View.VISIBLE);
+//            return false;
+//        }
+
+        textChannelError.setVisibility(View.GONE);
+        return true;
+    }
+
+    private void showLivestreamingOptions() {
+        progress.setVisibility(View.GONE);
+        livestreamPrecheckView.setVisibility(View.GONE);
+        livestreamOptionsView.setVisibility(View.VISIBLE);
+
+        checkCanStreamOnChannel((Claim) selectChannelSpinner.getSelectedItem());
+    }
+
+    private void updateChannelList(List<Claim> channels) {
+        selectChannelSpinner.setAdapter(new InlineChannelSpinnerAdapter(this, R.layout.spinner_item_channel, channels));
+
+        if (channels.size() > 0) {
+            showLivestreamingOptions();
+        } else {
+            progress.setVisibility(View.GONE);
+            showPrecheckError(getString(R.string.precheck_need_channel));
+        }
+    }
+
+    private void fetchChannels() {
+        if (Lbry.ownChannels != null && Lbry.ownChannels.size() > 0) {
+            updateChannelList(Lbry.ownChannels);
+            return;
+        }
+
+        ClaimListTask task = new ClaimListTask(Claim.TYPE_CHANNEL, null, new ClaimListResultHandler() {
+            @Override
+            public void onSuccess(List<Claim> claims) {
+                Lbry.ownChannels = new ArrayList<>(claims);
+                updateChannelList(Lbry.ownChannels);
+            }
+
+            @Override
+            public void onError(Exception error) {
+                if (error != null) {
+                    showError(error.getMessage());
+                }
+                progress.setVisibility(View.GONE);
+                showPrecheckError(getString(R.string.precheck_error_loading_channels));
+            }
+        });
+        task.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+    }
+
+    private String createStreamKey(String signature, String signingTs) {
+        String hexData = Helper.toHexString(selectedChannel.getName());
+        return selectedChannel.getClaimId()
+                + "?d=" + hexData
+                + "&s=" + signature
+                + "&t=" + signingTs;
+    }
+
+    private void showError(String message) {
+        Snackbar.make(findViewById(R.id.livestream_main), message, Snackbar.LENGTH_LONG).
+                setBackgroundTint(Color.RED).setTextColor(Color.WHITE).show();
+    }
+
+    private boolean isDarkMode() {
```

Comment:
This preference is now a little bit more complex as it could be "Use system setting". Check the implementation on one of the latest commits

Reply:
See getDarkModeSetting() method on MainActivity.

For Android 10+, dark mode could be Light/Night/System, being "Light" the default one; this setting is a string. On previous versions, it is a boolean for light/dark and default is "false" -meaning Light-.

Reply:
Yes, I found that code, but what I'm saying is that the UI seems to handle light/dark mode fine even without the code at https://github.com/OdyseeTeam/odysee-android/pull/167/files#diff-d31a44933ced9a1e3d8d3fd3123fbc7607c595729a97d9daf98e0e63cb4f0aadR114-R117

Reply:
@kekkyojin Is it necessary to add the light/dark/system handling code to GoLiveActivity?

The UI (including the status bar) seems to adapt properly even when I switch to dark mode while the app is running.